packages/wrangler/src/type-generation/index.ts

```diff
@@ -392,7 +446,32 @@ export async function generateEnvTypes(
 
 	if (configToDTS.services) {
 		for (const service of configToDTS.services) {
-			envTypeStructure.push([constructTypeKey(service.binding), "Fetcher"]);
+			const entrypointFile =
+				service.service !== entrypoint?.name
+					? serviceEntries?.get(service.service)?.file
+					: entrypoint?.file;
```

Comment:
Similarly just compute `serviceEntrypoint` here and reuse below.

Reply:
Fixed in a7f6a2b65